spinn_front_end_common/interface/interface_functions/local_tdma_builder.py

```diff
@@ -182,9 +182,7 @@ def _auto_config_times(
             n_slots = int(math.ceil(n_cores / app_machine_quantity))
             clocks_per_phase = (
                 int(math.ceil(overall_clocks_available / n_phases)))
-            # Note: The plus 1 ensures the last core finishes, if it's the
-            # worst in terms of n keys to transmit
-            clocks_between_cores = clocks_per_phase / (n_slots + 1)
+            clocks_between_cores = clocks_per_phase / n_slots
```

Comment:
Was the previous comment / code here incorrect, and the last core can never be the worst in term of n keys to transmit?

Reply:
The issue is that this is "just in case" rather than being anything else.  So it gives you one extra slot, but at the cost that all slots are smaller.  This ends up with things overrunning slots anyway where they might have been OK in the usual case.  It also complicates things as it adds a slot behind the scenes, meaning that your own manual calculations end up out, and you then have to work out why... it took me a while to find this and fix it!  It is worth noting that much of the code in these branches now doesn't use this at all for neurons, which appears to actually work better!  I believe that this was mostly of concern for Poisson sources (where it is still used I think) but neuron models tend to sort themselves out anyway.  Testing has shown that you don't seem to lose spikes without this in place in any case.